engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/StaticAnalysisTest.java

```diff
@@ -0,0 +1,138 @@
+package org.enso.compiler.test;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.logging.Level;
+import org.enso.common.LanguageInfo;
+import org.enso.common.MethodNames;
+import org.enso.common.RuntimeOptions;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.Diagnostic;
+import org.enso.compiler.core.ir.DiagnosticStorage;
+import org.enso.compiler.core.ir.Expression;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.core.ir.module.scope.definition.Method;
+import org.enso.editions.LibraryName;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.test.InterpreterContext;
+import org.enso.pkg.QualifiedName;
+import org.graalvm.polyglot.Source;
+import scala.jdk.javaapi.CollectionConverters;
+
+public abstract class StaticAnalysisTest {
+
+  /**
+   * The interpreter context is needed here as it ensures initialization of everything needed to
+   * perform imports resolution, including PackageRepository.
+   *
+   * <p>Ideally, the tests for the static analysis capabilities of the compiler should _not_ depend
+   * on the Graal runtime context, as they should be runnable in other contexts - i.e. in a Visual
+   * Studio Code language server.
+   */
+  private final InterpreterContext interpreterContext =
+      new InterpreterContext(
+          (builder) ->
+              builder
+                  .option(RuntimeOptions.ENABLE_STATIC_ANALYSIS, "true")
+                  .option(RuntimeOptions.LOG_LEVEL, Level.INFO.getName())
+                  .option(RuntimeOptions.LOG_LEVEL, Level.SEVERE.getName())
+                  .out(OutputStream.nullOutputStream())
+                  .err(OutputStream.nullOutputStream()));
+
+  private final EnsoContext langCtx =
+      interpreterContext
+          .ctx()
+          .getBindings(LanguageInfo.ID)
+          .invokeMember(MethodNames.TopScope.LEAK_CONTEXT)
+          .asHostObject();
+
+  protected Module compile(Source src) {
```

Comment:
Right. I think that these just did not exist when I wrote this, and I did not notice now that this was introduced - thanks for letting me know!

Reply:
After further inspection, my `compile` is doing more than the `ContextUtils::compile` - it is also registering the modules in scope so that they can be imported - which is crucial for tests that test behaviour of type inference across different modules. I guess perhaps I can try porting my additional 'features' to `ContextUtils`.

Reply:
I have kept my `compile` separate because IMO it differs more than is similar, because of the additional module management.

@Akirathan If you still think it should be merged then please let me know, we will find a way.